elastic/kibana
# [ECO][Inventory v2] Ad hoc data view: Add get entities definition endpoint using sources

Closes #202298 

This PR changes the way we get the entity index patterns to v2. It creates an endpoint part of the inventory API which returns the index patterns by entity type.

## Testing

### Test the endpoint: 
- Open Dev tools and add
` GET kbn:/internal/inventory/entity/definitions/sources`
- Response: 

![image](https://github.com/user-attachments/assets/3346c36e-dbc2-4e56-9ed6-d3d3a8f7d1a5)


### Test in the UI
- After the previous steps add some host data (oblt cluster / metricbeat) or use synthtrace (for example use `node scripts/synthtrace infra_hosts_with_apm_hosts --scenarioOpts.numInstances=10` or `node scripts/synthtrace logs_traces_hosts.ts`)
- Go to Inventory and expand the host group
- Click on the actions button for any host and click on the Discover link
- The correct dataview should be selected based on the index patterns in the source definition
The same can be done for other entity types
- Test the search bar as well (the suggestions should be visible) and now we should have 1 request to get the sources (instead of doing it on click)


https://github.com/user-attachments/assets/93b5ac6c-9d64-44e0-b26e-6133477e0840
